lib/modules/manager/npm/extract/common/package-file.ts

```diff
@@ -147,3 +148,18 @@ export function extractPackageJson(
     },
   };
 }
+
+export function hasPackageManager(content: string): boolean {
+  logger.trace(`npm.hasPackageManager from package.json`);
+  try {
+    const packageJsonResult = PackageJson.parse(content);
+
+    return (
+      is.nonEmptyString(packageJsonResult?.packageManager?.name) &&
+      is.nonEmptyString(packageJsonResult?.packageManager?.version)
+    );
+  } catch {
+    logger.debug(`Invalid JSON`);
+  }
```

Comment:
use `PackageJson.safeParse` so you don't need a try-catch

maybe this can be refactored / reused

https://github.com/renovatebot/renovate/blob/b084fe8faa248a2890fa2cdf4ab32648f6e929f5/lib/modules/manager/npm/post-update/utils.ts#L17-L29

Reply:
@viceice here is the PR for the later reuse https://github.com/renovatebot/renovate/pull/38114

Reply:
merged